File: apps/api-server/src/main/java/com/audioguide/entity/Poi.java
package com.audioguide.entity;

import com.audioguide.enums.PoiStatus;
import jakarta.persistence.*;
import lombok.*;
import lombok.experimental.FieldDefaults;

import java.time.LocalDateTime;

@Entity
@Table(name = "poi")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@FieldDefaults(level = AccessLevel.PRIVATE)
public class Poi {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    Integer id;

    @Column(nullable = false)
    Integer shopId;

    @Column(nullable = false)
    String name;

    @Column(columnDefinition = "TEXT")
    String description;

    @Column(nullable = false)
    String address;

    Double lat;

    Double lng;

    String region;

    String category;

    Integer ownerId;

    String ownerName;

    String coverImage;

    @Column(unique = true)
    String qrCode;

    @Column(nullable = false)
    Boolean riskFlag;

    Integer riskScore;

    String rejectionReason;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    PoiStatus status;

    @Column(nullable = false)
    LocalDateTime createdAt;

    @Column(nullable = false)
    LocalDateTime updatedAt;
}

File: apps/customer-web/src/types/poi.ts
export type PoiStatus = "DRAFT" | "PUBLISHED" | "FLAGGED" | "HIDDEN";

export type PoiCategoryKey =
    | "hai_san"
    | "lau"
    | "do_nuong"
    | "com"
    | "pho"
    | "giai_khat";

export type PoiSource = "api" | "manual";

export type PoiResponse = {
    id: number;
    shopId?: number;
    name: string;
    description?: string;
    address?: string;
    lat?: number;
    lng?: number;
    region?: string;
    category?: string;
    ownerId?: number;
    ownerName?: string;
    coverImage?: string;
    qrCode?: string;
    riskFlag: boolean;
    riskScore?: number;
    rejectionReason?: string;
    status: PoiStatus;
    createdAt: string;
    updatedAt: string;
};

export type MapPoi = {
    id: string;
    name: string;
    address?: string;
    description?: string;
    lat: number;
    lng: number;
    category?: string;
    categoryKey: PoiCategoryKey | null;
    status: PoiStatus;
    source: PoiSource;
};
